ajax_select/__init__.py

```diff
@@ -13,6 +13,9 @@
 from django.utils.encoding import force_text
 from django.utils.html import escape
 from django.utils.translation import ugettext_lazy as _
+from django.utils.module_loading import autodiscover_modules
```

Comment:
also if AJAX_LOOKUP_CHANNELS is not set and it tries to do autodiscovery
then it is on < 1.7 and cannot do autodiscovery it should throw an
ImproperlyConfigured.

"AJAX_LOOKUP_CHANNELS is not set and cannot do app autodiscovery unless
Django version is  >= 1.7"



On Thu, Jun 25, 2015 at 6:55 PM Chris Sattinger <crucialfelix@gmail.com>
wrote:

> That's a bit esoteric and confusing. I've had to read this paragraph
> several times and I still barely get it.
>
> so as it is, this is a breaking change. updating will break your site.
>
> why not call autodiscover iff settings.AJAX_LOOKUP_CHANNELS is not set ?
>
> if it is set, then load them as it has always loaded them.
>
>
>
> On Thu, Jun 25, 2015 at 4:53 PM Tony Morrow <notifications@github.com>
> wrote:
>
>> In ajax_select/__init__.py
>> <https://github.com/crucialfelix/django-ajax-selects/pull/129#discussion_r33263336>
>> :
>>
>> > @@ -13,6 +13,9 @@
>> >  from django.utils.encoding import force_text
>> >  from django.utils.html import escape
>> >  from django.utils.translation import ugettext_lazy as _
>> > +from django.utils.module_loading import autodiscover_modules
>>
>> The user doesn't call autodiscover, it is called by the configuration
>> module, apps.py, by subclassing AppConfig and overriding the ready()
>> method. The user indirectly calls autodiscover() by using the default
>> configuration. To use the other configuration without autodiscover(), the
>> user would enter 'ajax_select.SimpleAjaxSelectConfig' in the INSTALLED_APPS
>> dict instead of 'ajax_select'. The default app configuration is set in the
>> *init*.py. This is a Django 1.7+ feature (
>> https://docs.djangoproject.com/en/1.8/ref/applications/), so this also
>> breaks 1.5/1.6 compatibility. I can bring a simpler version of the
>> autodiscover_modules() functionality into the ajax_select app, though, in
>> order to support 1.5/1.6. I guess I also forgot to mention that this, too,
>> is a 1.5/1.6 breaking feature!
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/crucialfelix/django-ajax-selects/pull/129/files#r33263336>
>> .
>>
>